content/presence-occupancy/presence.textile

```diff
@@ -0,0 +1,702 @@
+/* request a wildcard token */
+AblyRest rest = new AblyRest("{{API_KEY}}");
+TokenParams tokenParams = new TokenParams() { ClientId = "*"};
+ClientOptions options = new ClientOptions();
+options.TokenDetails = await rest.Auth.RequestTokenAsync(tokenParams, null);
+
+AblyRealtime realtime = new AblyRealtime(options);
+IRealtimeChannel channel = realtime.Channels.Get("realtime-chat");
+
+channel.Presence.Subscribe(member => {
+  Console.WriteLine(member.ClientId + " entered realtime-chat");
+});
+
+await channel.Presence.EnterClientAsync("Bob", null); /* => Bob entered realtime-chat */
+await channel.Presence.EnterClientAsync("Mary", null); /* => Mary entered realtime-chat */
+```
+
+```[objc]
+ARTRest* rest = [[ARTRest alloc] initWithKey:@"{{API_KEY}}"];
+// request a wildcard token
+ARTTokenParams *tokenParams = [[ARTTokenParams alloc] initWithClientId:@"*"];
+[rest.auth requestToken:tokenParams withOptions:nil callback:^(ARTTokenDetails *tokenDetails,
+                                                              NSError *error) {
+    ARTRealtime *realtime = [[ARTRealtime alloc] initWithToken:tokenDetails.token];
+    ARTRealtimeChannel *channel = [realtime.channels get:@"realtime-chat"];
+
+    [channel.presence subscribe:ARTPresenceEnter callback:^(ARTPresenceMessage *member) {
+        NSLog(@"%@ entered realtime-chat", member.clientId);
+    }];
+
+    [channel.presence enterClient:@"Bob" data:nil]; // prints 'Bob entered realtime-chat'
+    [channel.presence enterClient:@"Mary" data:nil]; // prints 'Mary entered realtime-chat'
+}];
+```
+
+```[swift]
+let rest = ARTRest(key: "{{API_KEY}}")
+// request a wildcard token
+rest.auth.requestToken(ARTTokenParams(clientId: "*"), withOptions: nil) { tokenDetails, error in
+    let realtime = ARTRealtime(token: tokenDetails!.token)
+    let channel = realtime.channels.get("realtime-chat")
+
+    channel.presence.subscribe(.enter) { member in
+        print("\(member.clientId) entered realtime-chat")
+    }
+
+    channel.presence.enterClient("Bob", data: nil) // prints 'Bob entered realtime-chat'
+    channel.presence.enterClient("Mary", data: nil) // prints 'Mary entered realtime-chat'
+}
+```
+
+h2(#retrieve-presence-members). Retrieve presence members
+
+The @Presence@ object exposes a <span lang="default">@get@</span><span lang="csharp">@Get@</span> method allowing a client to retrieve an array of all members currently present on the channel. 
+
+h3(#realtime-member-list). Obtaining the presence list with the Realtime API
+
+The Ably client is responsible for keeping track of the presence set from the time that the channel is attached; an up to date presence set is pushed to the client following attach and the presence set is updated on each subsequent presence event. Thus "<span lang="default">@get@</span><span lang="csharp">@Get@</span>":/api/realtime-sdk/presence#get returns the already-known presence set retained in memory and does not trigger a new request to the Ably service.
```

Comment:
Do you think we need 2 separate sections for the difference between REST and realtime, or it would work with the introductory text explaining the difference and them sharing a code block?